src/core/processing-queue.js

```diff
@@ -61,13 +61,19 @@ function processTaskQueue( start ) {
 
 		if ( !defined.setTimeout || config.updateRate <= 0 || elapsedTime < config.updateRate ) {
 			const task = taskQueue.shift();
-			Promise.resolve( task() ).then( function() {
+			const processNextTaskOrAdvance = () => {
 				if ( !taskQueue.length ) {
 					advance();
 				} else {
 					processTaskQueue( start );
 				}
-			} );
+			};
+			const throwAndAdvance = ( err ) => {
+				setTimeout( advance );
+				throw err;
+			};
+
+			Promise.resolve( task() ).then( processNextTaskOrAdvance, throwAndAdvance );
```

Comment:
You want to add some comments here on what will happen if the rejected promise is not handled?